### PR TITLE
Fix wrong position for color dialog and sticky popups when styling analysis (+ metrics fix)

### PR DESCRIPTION
**Pull request by VictorVelarde**
### Related to https://github.com/CartoDB/support/issues/1649, https://github.com/CartoDB/support/issues/1673 and https://github.com/CartoDB/cartodb/issues/14183

This issue deals with 2 blocks:
A. style issues
B. metric issues

The most important part is the first one (metrics is a little *bonus*).

###  Context
There are issues regarding to color picker being placed in a wrong position (#1649 and #1673) and also information popups getting stuck at the same time (#1673).

The situation has been clearly reproduced for #1649, which is a map with an analysis (_Add Columns from Second Dataset_). For the second case #1673, we didn't have confirmation but it seems a very similar case with another analysis.

### Cause
When dealing with `analyses layers` (not simple layers) an event can be raised, in the interaction with carto.js, and that was causing an unexpected re-render in the style tab, breaking its color dialog (and sometimes its popups). The offending handler was this Line 118 in `style-content-view.js`:
```javascript
this.listenTo(this._queryGeometryModel, 'change:status', this.render);
```
I have checked it and it was added here (before the size & color project): [Little refactor improvements](https://github.com/CartoDB/cartodb/commit/d4c2b0e12ecf37b52f4cfe8c1eb934be35f05c55),

I have removed it, tests pass, and it seems that it solves our issues, but I honestly can't see the point why it was there in the first place. I made some tests on non-geocoded and empty layers, and it seems they are not affected... @ivanmalagon @elenatorro In what situation could it be useful?

### Acceptance A.
Try at staging with https://displaced-dialog.carto-staging.com/
- [ ] Test one map with an analysis (and simple) layers, altering its color fill properties several times (`fixed mode`).
- [ ] Repeat testing with `by value mode`
- [ ] Expected: emerging dialog to define color has a correct position and popups disappear.

---
### Acceptance B.
- [ ] Open any map
- [ ] Edit the style
- [ ] Open the console / network in devtools to see there is no error regarding to metrics

### CODE REVIEW

**Comment by VictorVelarde** [comment]
retest this please

**Comment by VictorVelarde** [comment]
retest this please

File: lib/assets/javascripts/builder/editor/style/style-content-view.js [hunk 1]
```diff
@@ -115,7 +115,6 @@ module.exports = CoreView.extend({
     this.listenTo(this._layerContentModel, 'change:state', this._checkEditorModel);
     this.listenTo(this._overlayModel, 'change:visible', this._toggleOverlay);
     this.listenTo(this._viewState, 'change:isDataFiltered', this.render);
-    this.listenTo(this._queryGeometryModel, 'change:status', this.render);
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply lib/assets/javascripts/builder/editor/style/style-content-view.js]: It is, a very tricky one indeed

[the remark this answers is not in the document]
Comment (VictorVelarde) [reply lib/assets/javascripts/builder/editor/style/style-content-view.js]: 😈

**Comment by VictorVelarde** [comment]
@ramiroaznar This is ready for acceptance! Steps 🔝

File: lib/assets/javascripts/builder/components/metrics/metrics-types.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var METRIC_TYPES = {
```
Comment (javitonino) [inline lib/assets/javascripts/builder/components/metrics/metrics-types.js]: I'd add a comment here telling to keep this synced with `events.rb`. And viceversa, a comment in `events.rb` telling us to check this file.

Hopefully that should avoid some problems.

Comment (VictorVelarde) [reply lib/assets/javascripts/builder/components/metrics/metrics-types.js]: thx @javitonino, I will

Comment (javitonino) [reply lib/assets/javascripts/builder/components/metrics/metrics-types.js]: :heart:

**Comment by VictorVelarde** [comment]
retest this please

**Comment by ramiroaznar** [comment]
LGTM!

I see a google error in the console. But I do not think it is related to the metrics:

image

**Comment by VictorVelarde** [comment]
Our metrics point to our backend, with urls like: `.../api/v3/metrics`. This googletagmanager 404 seems unrelated, and it looks like something not available at staging. 

So we're gonna merge and deploy this...

**Pull request merged**